src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -588,51 +601,63 @@ internal static string GetClientMutexName(string pipeName)
         }
 
         /// <summary>
-        /// Gets the value of the temporary path for the current environment assuming the working directory
-        /// is <paramref name="workingDir"/>.  This function must emulate <see cref="Path.GetTempPath"/> as 
-        /// closely as possible.
+        /// Gets the value of the temporary path for the provided environment settings. This behavior
+        /// is OS specific.
+        ///   - On Windows it seeks to emulate Path.GetTempPath as closely as possible with 
+        ///     provided working directory.
         /// </summary>
         internal static string? GetTempPath(string? workingDir)
         {
-            if (PlatformInformation.IsUnix)
+            return PlatformInformation.IsUnix
+                ? getTempPathLinux()
+                : getTempPathWindows(workingDir);
+
+            static string? getTempPathLinux()
             {
                 // Unix temp path is fine: it does not use the working directory
                 // (it uses ${TMPDIR} if set, otherwise, it returns /tmp)
+                //
+                // https://github.com/dotnet/roslyn/issues/65415 tracks moving to a directory under
+                // $XDG_RUNTIME_DIR if it is set using appropriate fall backs as necessary.
+                // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
```

Comment:
I don't understand this comment and given the issue referenced here, could we just say "to a per-user directory"?